### PR TITLE
Add support for LogicBone board

### PR DESCRIPTION
**Pull request by VioletEternity**
Based on the platform file by Owen Kirby: https://github.com/oskirby/logicbone-tests/blob/b8254bc7a177077abc939152cf0f3d3648f6a925/nmigen/logicbone_platform.py

Also based on the platform file by Katherine Temkin: https://github.com/greatscottgadgets/luna/blob/08b035c9c2b053d7edffb0d220d948b5c2ca927e/luna/gateware/platform/logicbone.py

### CODE REVIEW

File: amaranth_boards/logicbone.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        Connector("P9", 0, """
+        -   -   -   -   -   -   -   -   -   -   -   A11 B11 A10 C10 A9
+        B9  C11 A8  -   -   D9  C8  B8  A7  A6  B6  D8  C7  D7  C6  D6
+        -   -   -   -   -   -   -   -   -   B10 E10 -   -   -
+        """),
+    ]
+
+    def toolchain_prepare(self, fragment, name, **kwargs):
+        overrides = dict(ecppack_opts="--compress --spimode qspi --freq 38.8")
+        overrides.update(kwargs)
```
Comment (modwizcode) [inline amaranth_boards/logicbone.py:102]: As far as I can tell this line isn't needed.

**Review by modwizcode — CHANGES_REQUESTED** [review]
One tiny change, otherwise LGTM.

File: amaranth_boards/logicbone.py (hunk 1 above)
Comment (VioletEternity) [reply amaranth_boards/logicbone.py:102]: Rather, the next line should be: `return super().toolchain_prepare(fragment, name, **overrides)`.

**Review by modwizcode — DISMISSED** [review]
LGTM. Thank you for the contribution!

**Review by modwizcode — APPROVED** [review]

File: amaranth_boards/logicbone.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            attrs=Attrs(IO_TYPE="LVCMOS33")
+        ),
+        DirectUSBResource("usb", 1,
+            d_p="R1",  d_n="T1",  pullup="T2",
+            attrs=Attrs(IO_TYPE="LVCMOS33")
+        ),
+
+        *LEDResources(pins="D16 C15 C13 B13", attrs=Attrs(IO_TYPE="LVCMOS33"), invert=True),
+
+        *SwitchResources(pins="U2", attrs=Attrs(IO_TYPE="LVCMOS33")),
```
Comment (whitequark) [inline amaranth_boards/logicbone.py:46]: Should be:
```suggestion
        *ButtonResources(pins="U2", attrs=Attrs(IO_TYPE="LVCMOS33"), invert=True),
```

**Review by whitequark — APPROVED** [review]
LGTM

**Comment by whitequark** [comment]
Thank you!

**Pull request merged**